Modules as tables on scrabble-score and roman-numerals

etandel opened the pull request:
Ever since the extinguishing of `module()` the "lunatic" way of making modules is to return a table with the module's content.

kytrinyx commented:
Would you mind rebasing onto master, please?

etandel commented:
Done!

kytrinyx commented:
Thanks! I'm leaving this for @aarti to merge, since she has the high-level overview of the track.

aarti commented:
@etandel  so you are converting to a local function and returning it wrapped in a table. I read about the new way on two blog posts  http://hisham.hm/2014/01/02/how-to-write-lua-modules-in-a-post-module-world/ http://blog.separateconcerns.com/2014-01-03-lua-module-policy.html, but I am not clear on your change.

etandel commented:
To be honest, I'm not sure what you expect me to explain, so I'll just explain it all. =P

Ever since I know Lua (5.1), modules are expressed as tables, since those are, of course, the only sensible [1] way of aggregating multiple values. At the time, the _de jure_ way of doing this was using the `module()` function, which also did a lot of magic that allowed, amongst other things:
 - multiple modules on the same file;
 - "global" variables to be automatically exported inside the module (instead of polluting the actual global namespace);
 - cyclic dependencies.

However, mostly because of that magic, people began to consider `module()` to be harmful, so it was deprecated on 5.2 and removed from 5.3.

Since then, the _de facto_ way of creating modules is to explicitly put the values to be exported on a table and `return` that table at the end of the file.

The tests for these exercises were expecting the function to be exported by itself, which breaks both the old and the new module pattern. While it doesn't make much difference for simple code like this, the whole point of exercism itself is to help newcomers to learn the idioms of the language, so I changed the code to use the new module idiom, just like the rest of this track's exercises.

An example of code that uses this pattern is luarocks itself: https://github.com/keplerproject/luarocks/blob/master/src/luarocks/upload.lua

[1] It is possible to use closures and their internal stacks as data structures for holding multiple lua values but it's nonsensical, though extremely fast for some cases.
